### ISSUE TITLE
v.2.6.4 crashes, memory error.

### ISSUE THREAD

**Issue by CEHitchens** [johnfactotum/foliate#1066]
**Describe the bug**
Foliate crashes after a few seconds.

**To Reproduce**
1. Open an epub. Whether via file manager (Nemo) or from within Foliate makes no difference
2. Either browse through pages or wait about 10 seconds.
3. Witness brief CPU load increase
4. Program crashes
5, Output after having launched Foliate from a terminal by: com.github.johnfactotum.Foliate --verbose

`$ com.github.johnfactotum.Foliate --verbose
Gjs-Message: 10:13:55.952: JS WARNING: [resource:///com/github/johnfactotum/Foliate/js/main.js 23]: Requiring WebKit2 but it has 2 versions available; use imports.gi.versions to pick one
Gjs-Message: 10:13:56.560: JS LOG: ready
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: doubleClickTime =
                    400
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: imgEventType = "click"
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: windowSize = 1487
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: windowHeight = 1074
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: zoomLevel = 1
Gjs-Message: 10:13:56.560: JS LOG: run_javascript: enableFootnote = true
Gjs-Message: 10:13:56.561: JS LOG: run_javascript: autohideCursor = false
Gjs-Message: 10:13:56.561: JS LOG: run_javascript: open(
                    decodeURI("file:///<PATH_TO_EBOOK>/<EBOOK_NAME>.epub"),
                    decodeURI("Author...
Gjs-Message: 10:13:56.773: JS LOG: book-ready
Gjs-Message: 10:13:56.773: JS LOG: run_javascript: JSON.stringify(book.package.metadata)
Gjs-Message: 10:13:56.773: JS LOG: run_javascript: JSON.stringify(book.navigation.toc)
Gjs-Message: 10:13:56.776: JS LOG: saving file uri
Gjs-Message: 10:13:56.777: JS LOG: run_javascript: loadLocations(["epubcfi(/6/4[title]!/4,/12/1:0,/16/4/1:7)","epubcfi(/6/6[copyright]!/4,/6/4/1:0,/22/1:4)","epubcfi(/6/6[copyright]!/4,/22/1:5,/28/1:30)","epubcfi(/6/8[dedi]!/4/4/4,/1:0,/1:11)","epubcf...
Gjs-Message: 10:13:56.777: JS LOG: run_javascript: render()
Gjs-Message: 10:13:56.779: JS LOG: cover
Gjs-Message: 10:13:56.787: JS LOG: saving cover to /home/<USERNAME>/.cache/com.github.johnfactotum.Foliate/9781416588276.png
Gjs-Message: 10:13:56.796: JS LOG: locations-ready
Gjs-Message: 10:13:56.796: JS LOG: run_javascript: JSON.stringify(sectionMarks)

(com.github.johnfactotum.Foliate:222805): Gjs-WARNING **: 10:13:56.798: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
  main@resource:///com/github/johnfactotum/Foliate/js/main.js:476:24
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  @/usr/bin/com.github.johnfactotum.Foliate:9:17

Gjs-Message: 10:13:56.843: JS LOG: rendition-ready
Gjs-Message: 10:13:56.844: JS LOG: run_javascript: setStyle({"fontFamily":"Luebeck","fontSize":22.666666666666668,"fontWeight":600,"fontStyle":"normal","fontStretch":"normal","spacing":1.5,"margin":40,"maxWidth":1400,"usePublisherFont":false,"justify"...
Gjs-Message: 10:13:56.844: JS LOG: run_javascript: setupRendition()
Gjs-Message: 10:13:56.844: JS LOG: run_javascript: display('epubcfi(/6/22[ch05]!/4/92/9:69)')
Gjs-Message: 10:13:56.858: JS LOG: layout
Gjs-Message: 10:13:57.194: JS LOG: book-displayed
Gjs-Message: 10:13:57.294: JS LOG: relocated
Gjs-Message: 10:13:57.294: JS LOG: epubcfi(/6/22[ch05]!/4/92/9:69)
Gjs-Message: 10:13:57.777: JS LOG: writing to /home/USERNAME/.local/share/com.github.johnfactotum.Foliate/library/uri-store.json
Gjs-Message: 10:13:58.296: JS LOG: writing to /home/USERNAME/.local/share/com.github.johnfactotum.Foliate/9781416588276.json
Gjs-Message: 10:13:58.299: JS LOG: book list: updating 9781416588276
Gjs-Message: 10:13:58.299: JS LOG: book list: updating 9781416588276
Speicherzugriffsfehler (Speicherabzug geschrieben)
`

**Expected behavior**
Be able to read epubs without a crash, as I was able to do on the very same PC, earlier, which was before I upgraded from Linux Mint 20.3 Cinnamon.

**Screenshots**
None. But see CLI output above.

**Version:**
 - Foliate version: 2.6.4 (also tested 2.6.3)
 - OS/Distribution and version: Linux Mint 21.2
 - Desktop environment: Cinnamon
 - Installation method: deb packages, downloaded from Foliate's GitHub (flatpaks are way too big imho)

**Additional context**
Regarding dependencies:
- gjs/jammy-updates,now 1.72.4-0ubuntu0.22.04.1 amd64     is installed
- gtk4 (>= 4.10)  not available via LM's repos
- gir1.2-adw-1/jammy-updates,now 1.1.7-0ubuntu0.22.04.1 amd64      is installed
- gir1.2-webkit-6.0/jammy-updates,jammy-security,now 2.40.4-0ubuntu0.22.04.1 amd64     is installed

I might fully reinstall Linux Mint 21.2 Cinnamon at some point.

**Comment by abadalen** [comment]
I solved it by running this command: `export G_SLICE=always-malloc evolution`

**Comment by CEHitchens** [comment]
It was suggested to me that I should try running: "export G_SLICE=always-malloc evolution" prior to running Foliate. Unfortunately that did not solve the issue in my case.

Ultimately, I wiped the Linux Mint 21.2 Cinnamon installation, which got upgraded from LM20.3. Since Foliate did not crash/close itself in a LM21.2 Cinnamon USB live session, I installed a fresh LM21.2 Cinnamon >> Foliate is not crashing any more.

Conclusion: it is highly likely that something capsized during the OS upgrade.
This issue can be closed.
